drakrun/drakrun/draksetup.py

```diff
@@ -25,28 +25,39 @@
 FNULL = open(os.devnull, 'w')
 
 
-def install(storage_backend, disk_size, iso_path, zfs_tank_name, max_vms, unattended_xml):
-    logging.info("Ensuring that drakrun@* services are stopped...")
-    subprocess.check_output('systemctl stop \'drakrun@*\'', shell=True, stderr=subprocess.STDOUT)
+def find_default_interface():
+    routes = subprocess.check_output('ip route show default', shell=True, stderr=subprocess.STDOUT) \
+        .decode('ascii').strip().split('\n')
 
-    logging.info("Performing installation...")
+    for route in routes:
+        m = re.search(r'dev ([^ ]+)', route.strip())
 
-    if storage_backend != "qcow2":
-        raise RuntimeError("Not implemented yet")
+        if m:
+            return m.group(1)
 
+    return ''
+
+
+def detect_defaults():
     os.makedirs(ETC_DIR, exist_ok=True)
     os.makedirs(os.path.join(ETC_DIR, "configs"), exist_ok=True)
 
     os.makedirs(LIB_DIR, exist_ok=True)
     os.makedirs(os.path.join(LIB_DIR, "profiles"), exist_ok=True)
     os.makedirs(os.path.join(LIB_DIR, "volumes"), exist_ok=True)
 
-    conf_path = os.path.join(ETC_DIR, "config.ini")
     conf = configparser.ConfigParser()
-    conf.read(conf_path)
-
+    conf.read(os.path.join(ETC_DIR, "config.ini"))
     conf_patched = False
+
     minio_access_key = conf.get('minio', 'access_key').strip()
+    out_interface = conf.get('drakrun', 'out_interface').strip()
+
+    if not out_interface:
+        default_if = find_default_interface()
+        logging.info(f"Detected default network interface: {default_if}")
+        conf['drakrun']['out_interface'] = default_if
+        conf_patched = True
```

Comment:
Maybe use uniform interface to access config - `.get`
Additionally it provides `fallback` kwarg which would simplify the code logic

Reply:
This is actually a config write, not read. For reads, `.get` is already used everywhere.